tests/test_batch.py

```diff
@@ -1,71 +1,91 @@
 
-    @patch("sdk.Batch")
-    def test_create_batch(self, mock_batch):
-        self.sdk.create_batch(
-            serialized_sequence=self.pulser_sequence,
+        # Configure requests to use the local JSON files a response
+        kwargs["mock"].register_uri(
+            requests_mock.ANY, requests_mock.ANY, json=mock_response
         )
-        self.sdk._client._send_batch.assert_called_once()
+        self.requests_mock = kwargs["mock"]
+        self.sdk = SDK(client_id="my_client_id", client_secret="my_client_secret")
+        self.pulser_sequence = "pulser_test_sequence"
+        self.batch_id = 1
+        self.job_result = "result"
+        self.n_job_runs = 50
 
-    @patch("sdk.batch.Client")
-    def test_batch_add_job(self, mock_client):
-        batch_id = 2
-        mock_client._send_job.return_value = {
-            "runs": 1,
-            "batch_id": batch_id,
-            "id": 1,
-            "status": "",
-            "created_at": "",
-            "updated_at": "",
-            "errors": [],
-        }
-        self.batch = Batch(
-            complete=False,
-            created_at="",
-            updated_at="",
-            device_type="",
-            group_id=1,
-            id=batch_id,
-            user_id=1,
-            priority=1,
-            status="",
-            webhook="",
-            _client=mock_client,
-            sequence_builder=self.pulser_sequence,
-        )
-        job = self.batch.add_job(runs=50, variables={"var": 2})
-        self.batch._client._send_job.assert_called_once()
-        assert job.batch_id == batch_id
+    def test_create_batch(self):
+        with self.requests_mock:
+            batch = self.sdk.create_batch(
+                serialized_sequence=self.pulser_sequence,
+            )
+            assert batch.id == self.batch_id
+            assert batch.sequence_builder == self.pulser_sequence
 
-    @patch("sdk.batch.Job")
-    @patch("sdk.batch.Client")
-    def test_batch_declare_complete(self, mock_job, mock_client):
-        self.batch = Batch(
-            complete=False,
-            created_at="",
-            updated_at="",
-            device_type="",
-            group_id=1,
-            id=1,
-            user_id=1,
-            priority=1,
-            status="",
-            webhook="",
-            _client=mock_client,
-            sequence_builder=self.pulser_sequence,
-        )
-        self.batch.declare_complete(wait=True)
-        self.batch._client._complete_batch.assert_called_once()
+    def test_batch_add_job(self):
+        with self.requests_mock:
+            batch = self.sdk.create_batch(
+                serialized_sequence=self.pulser_sequence,
+            )
+            job = batch.add_job(
+                runs=self.n_job_runs,
+                variables={"Omega_max": 14.4, "last_target": "q1", "ts": [200, 500]},
+            )
+            assert job.batch_id == self.batch_id
+            assert job.runs == self.n_job_runs
```

Comment:
These tests are a bit artificial since we just check the value against the one in the fixture ut not sure we have anything meaningful to test anyway. At least we will detect exceptions.

Reply:
No I don't think the request args should appear in json. However, just thinking about it, what I think we could test here is actually test the body of the request that was sent, e.g. making sure the group_id was automatically set in the request body!

That's one of the "main" feature of this SDK so that's precisely what we should test

Same thing for job creation, make sure the batch id is properly set for the user